src/NuGet.Core/NuGet.Protocol/Resources/PackageMetadataResourceV3.cs

```diff
@@ -38,37 +55,240 @@ public class PackageMetadataResourceV3 : PackageMetadataResource
+            {
+                Index = index,
+                CacheContext = httpSourceCacheContext
+            };
+        }
+
+        /// <summary>
+        /// Process RegistrationIndex 
+        /// </summary>
+        /// <param name="httpSource">Httpsource instance.</param>
+        /// <param name="rangeUri">Paged registration index url address.</param>
+        /// <param name="packageId">PackageId for package we're checking.</param>
+        /// <param name="lower">Lower bound of nuget package.</param>
+        /// <param name="upper">Upper bound of nuget package.</param>
+        /// <param name="httpSourceCacheContext">SourceCacheContext for cache.</param>
+        /// <param name="log">Logger Instance.</param>
+        /// <param name="token">Cancellation token.</param>
+        /// <returns></returns>
+        private async Task<RegistrationPage> GetRegistratioIndexPageAsync(
+            HttpSource httpSource,
+            string rangeUri,
+            string packageId,
+            NuGetVersion lower,
+            NuGetVersion upper,
+            HttpSourceCacheContext httpSourceCacheContext,
+            ILogger log,
+            CancellationToken token)
+        {
+            var packageIdLowerCase = packageId.ToLowerInvariant();
+            var registrationPage = await httpSource.GetAsync(
+                            new HttpSourceCachedRequest(
+                                rangeUri,
+                                $"list_{packageIdLowerCase}_range_{lower.ToNormalizedString()}-{upper.ToNormalizedString()}",
+                                httpSourceCacheContext)
+                            {
+                                IgnoreNotFounds = true,
+                            },
+                            async httpSourceResult =>
+                            {
+                                return await ProcessRegistrationIndexAsync<RegistrationPage>(httpSourceResult, token);
+                            },
+                            log,
+                            token);
+
+            return registrationPage;
+        }
+
+        /// <summary>
+        /// Process RegistrationPage
+        /// </summary>
+        /// <param name="registrationPage">Nuget registration page.</param>
+        /// <param name="results">Used to return nuget result.</param>
+        /// <param name="range">Nuget version range.</param>
+        /// <param name="includePrerelease">Whether to include PreRelease versions into result.</param>
+        /// <param name="includeUnlisted">Whether to include Unlisted versions into result.</param>
+        private void ProcessRegistrationPage(
+            RegistrationPage registrationPage,
+            List<PackageSearchMetadataRegistration> results,
+            VersionRange range, bool includePrerelease,
+            bool includeUnlisted,
+            MetadataReferenceCache metadataCache)
+        {
+            foreach (var registrationLeaf in registrationPage.Items)
```

Comment:
Per [coding guideline](https://github.com/NuGet/NuGet.Client/blob/dev/docs/coding-guidelines.md#c-coding-style) 10, we shouldn't use `var` here.

Reply:
Fixed.